src/views/Unsubscribe/index.tsx

```diff
@@ -0,0 +1,174 @@
+
+import Page from '#components/Page';
+import {
+    AlertSubscriptionsUnsubscribeQuery,
+    AlertSubscriptionsUnsubscribeQueryVariables,
+    UnsubscriptionMutation,
+    UnsubscriptionMutationVariables,
+} from '#generated/types/graphql';
+import useAlert from '#hooks/useAlert';
+
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+const ALERT_SUBSCRIPTIONS = gql`
+    query AlertSubscriptionsUnsubscribe {
+        private {
+            id
+            userAlertSubscriptions {
+                items {
+                    id
+                    name
+                    isActive
+                    filterAlertAdmin1s
+                    filterAlertCountryId
+                }
+            }
+        }
+    }
+`;
+
+const UPDATE_SUBSCRIPTION = gql`
+    mutation Unsubscription(
+        $subscriptionId: ID!,
+        $data: UserAlertSubscriptionInput!,
+    ) {
+        private {
+            updateUserAlertSubscription(
+                id: $subscriptionId,
+                data: $data,
+            ) {
+                errors
+                ok
+                result {
+                    id
+                    isActive
+                }
+            }
+        }
+    }
+`;
+
+// eslint-disable-next-line import/prefer-default-export
+export function Component() {
+    const { subscriptionId } = useParams<{ subscriptionId?: string, token?: string }>();
+    const alert = useAlert();
+    const strings = useTranslation(i18n);
+    const [isErrored, setIsError] = useState(false);
+    const [isSubmitted, setIsSubmitted] = useState(false);
+
+    const {
+        data: alertSubscriptions,
+    } = useQuery<
+        AlertSubscriptionsUnsubscribeQuery,
+        AlertSubscriptionsUnsubscribeQueryVariables
+    >(
+        ALERT_SUBSCRIPTIONS,
+    );
+
+    const [
+        subscriptionUpdate,
+    ] = useMutation<
+        UnsubscriptionMutation,
+        UnsubscriptionMutationVariables
+    >(
+        UPDATE_SUBSCRIPTION,
+        {
+            onCompleted: (response) => {
+                const subscriptionResponse = response?.private.updateUserAlertSubscription;
+                if (!response) {
+                    return;
+                }
+                if (
+                    subscriptionResponse.ok
+                    && subscriptionResponse.result?.isActive === false
+                ) {
+                    setIsSubmitted(true);
+                } else {
+                    setIsError(true);
+                }
+            },
+            onError: () => {
+                alert.show(
+                    strings.unsubscriptionFailed,
+                    { variant: 'danger' },
+                );
+            },
+        },
+    );
+
+    const data = alertSubscriptions?.private?.userAlertSubscriptions?.items;
+
+    const handleUnsubscribe = useCallback((id: string) => {
+        const selectedSubscriptionDetails = data?.find(
+            (sub) => sub.id === id,
+        );
```

Comment:
Do we need this?

Reply:
We need it; else it will update the subscription admin1 & country.